src/colorizer/Collection.ts

```diff
@@ -124,18 +124,32 @@ export default class Collection {
    *
    * See `DatasetLoadResult` for more details.
    */
-  public async tryLoadDataset(datasetKey: string): Promise<DatasetLoadResult> {
+  public async tryLoadDataset(
+    datasetKey: string,
+    onLoadProgress?: (complete: number, total: number) => void
+  ): Promise<DatasetLoadResult> {
     console.time("loadDataset");
 
     if (!this.hasDataset(datasetKey)) {
       return { loaded: false, dataset: null, errorMessage: `Error: Collection does not have key ${datasetKey}.` };
     }
     const path = this.getAbsoluteDatasetPath(datasetKey);
     console.log(`Fetching dataset from path '${path}'`);
+
+    let totalLoadItems = 0;
+    let completedLoadItems = 0;
+    const onLoadStart = (): void => {
+      totalLoadItems++;
+    };
+    const onLoadCompleted = (): void => {
+      completedLoadItems++;
+      onLoadProgress?.(completedLoadItems, totalLoadItems);
+    };
+
     // TODO: Override fetch method
     try {
       const dataset = new Dataset(path);
-      await dataset.open();
+      await dataset.open(undefined, onLoadStart, onLoadCompleted);
```

Comment:
This `undefined` here makes me cringe a little... It's an argument that overrides the fetching method `dataset.open` uses, and in this case I don't want to touch. (It's mostly used in unit tests to spoof `fetch` operations.)

Open to input on if I can make this cleaner.

Reply:
one thing would just be to make it not optional and send in the loader you want here. the other way would be to send in a config object that has the optional settings, instead of a list of arguments, and the config object could have just the onLoadStart and onLoadCompleted, and you wouldn't need to set anything to undefined

Reply:
Went the route of including a config object!